api/python/provisioner/inputs.py

```diff
@@ -877,3 +879,60 @@ def is_iso(self):
 @attr.s(auto_attribs=True)
 class SWUpgradeRepo(SWUpdateRepo):
     _param_di = param_spec['swupgrade/repo']
+    # file path to base directory for SW upgrade
+    _target_build: str = attr.ib(default=None)
+    # parameter to enable or disabled repository
+    _enabled: bool = attr.ib(default=False)
+
+    @property
+    def pillar_key(self):
+        # the local root pillar key for 'swupgrade' installation type
+        return self.release
+
+    @property
+    def pillar_value(self):
+        if self.is_special():
+            res = {
+                f"{repo}": None
+                for repo in (config.OS_ISO_DIR,
+                             config.CORTX_ISO_DIR,
+                             config.CORTX_3RD_PARTY_ISO_DIR,
+                             config.CORTX_PYTHON_ISO_DIR)
+            }
+            res[self.release] = None
+
+            return res
+        elif self.is_remote():
+            return self.source
```

Comment:
in case of remote it would become a `target_build` and we need to setup few repos from that as well (cortx, 3rd parties, os, ...). You may treat that like ISO mounted on the host server.

Please create a follow up task for remote urls support.